prefect_azure/credentials.py

```diff
@@ -382,3 +382,30 @@ def example_get_workspace_flow():
         )
 
         return workspace
+
+
+class ContainerInstanceCredentials(Block):
+    """
+    Block used to manage Azure Container Instances authentication. Stores Azure Service
+    Principal authentication data.
+
+    Args:
+        client_id: The service principal client ID.
+        tenant_id: The service principal tenant ID.
+        client_secret: The service principal client secret.
+
+    Example:
+        Load stored Azure Blob Storage credentials:
+        ```python
+        from prefect_azure import ContainerInstanceCredentials
+        azure_credentials_block = ContainerInstanceCredentials.load("BLOCK_NAME")
+        ```
+    """
+
+    _block_type_slug = "azure-container-credentials"
+    _block_type_name = "Azure Container Credentials"
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/6AiQ6HRIft8TspZH7AfyZg/39fd82bdbb186db85560f688746c8cdd/azure.png?h=250"  # noqa
+
+    client_id: SecretStr
+    tenant_id: SecretStr
```

Comment:
Do these need to be Secret?
```suggestion
    client_id: str
    tenant_id: str
```